src/servers/src/elasticsearch.rs

```diff
@@ -132,19 +133,18 @@ async fn do_handle_bulk_api(
     // The `schema` is already set in the query_ctx in auth process.
     query_ctx.set_channel(Channel::Elasticsearch);
 
-    let db = params.db.unwrap_or_else(|| "public".to_string());
+    let db = query_ctx.current_schema();
```

Comment:
We can support both query parameters and headers, giving preference to headers when both are present.